File: implementations/flutter/design_patterns_flutter/lib/patterns/behavioral/chain_of_responsibility/soporte_nivel_2.dart
import 'handler.dart';
import 'types.dart';

class SoporteNivel2 extends Handler {
  @override
  HandleResult handle(Ticket req, {List<String>? trail}) {
    final t = trail ?? <String>[];
    push(t);
    if (req.category == Category.technical &&
        severityRank(req.severity) >= severityRank(Severity.medium)) {
      return HandleResult(
        handledBy: 'SoporteNivel2',
        message: 'Asignado a ingeniero de Nivel 2.',
        trail: t,
      );
    }
    return super.handle(req, trail: t);
  }
}

File: implementations/react-native/design-patterns-rn/src/patterns/behavioral/chain_of_responsibility/soporte_nivel_2.ts
import { SupportHandler } from './base_handler';
import { rank } from './helpers_rank';
import type { HandleResult, Ticket } from './types';

export class SoporteNivel2 extends SupportHandler {
  handle(req: Ticket, trail: string[] = []): HandleResult {
    this.push(trail);
    if (req.category === 'technical' && rank[req.severity] >= rank['medium']) {
      return { handledBy: 'SoporteNivel2', message: 'Asignado a ingeniero de Nivel 2.', trail };
    }
    return super.handle(req, trail);
  }
}
